arch/xtensa/soc/esp32/esp32-mp.c

```diff
@@ -43,9 +44,40 @@ struct cpustart_rec {
 volatile struct cpustart_rec *start_rec;
 static void *appcpu_top;
 
+static struct k_spinlock loglock;
+
+/* Note that the logging done here is ACTUALLY REQUIRED FOR RELIABLE
```

Comment:
ouch, we got a bug open on this?

Reply:
No, but we suspect this is related to the ESP32 hardware revision.  Revision 0 has some issues with the UART FIFO, that were fixed in the Revision 1 (according to [this errata document](https://www.espressif.com/sites/default/files/documentation/eco_and_workarounds_for_bugs_in_esp32_en.pdf)).  It's likely that the ROM UART routines are not using the alternate FIFO addresses, and thus this makes no difference in some of the boards with a newer ESP32 chip.  I don't know yet what exactly are the differences that causes the need for this delay/usage of the UART, but this is one of the things that we can ask Espressif.

Maybe @igrr, @spritetm or @projectgus could shed some light here?